odoo/addons/base/tests/test_cloc.py

```diff
@@ -96,3 +96,40 @@ def test_parser(self):
             self.assertEqual(py_count, (8, 16))
         js_count = cl.parse_js(JS_TEST)
         self.assertEqual(js_count, (10, 17))
+
+    def test_ignore_auto_generated_SA(self):
+        server_action_values = {
+            'name': 'Merge Test',
+            'binding_view_types': 'list',
+            'state': 'code',
+            'code': "action = env['data_merge.record'].action_deduplicates(records)",
+            'model_id': self.env.ref('base.model_res_partner').id,
+        }
+        self.env['ir.actions.server'].create(server_action_values)
+        cl = cloc.Cloc()
+        cl.count_customization(self.env)
+        self.assertEqual(cl.code.get('odoo/studio', 0), 0, 'Data cleaning SA should not be counted in cloc')
+
+    def test_ignore_auto_generated_computed_field(self):
+        compute_function = """
+            results = self.env['%(model)s'].read_group([('%(field)s', 'in', self.ids)], ['%(field)s'], ['%(field)s'])
+            dic = {}
+            for x in results: dic[x['%(field)s'][0]] = x['%(field)s_count']
+            for record in self: record['%(count_field)s'] = dic.get(record.id, 0)
+        """ % {
+            'model': 'res.partner',
+            'field': 'invoice_count',
+            'count_field': 'invcoie_count_res_partner',
+        }
+        self.env['ir.model.fields'].create({
+            'name': 'x_invoice__res_partner_count',
+            'field_description': 'Invoice count',
+            'model': 'res.partner',
+            'model_id': self.env.ref('base.model_res_partner').id,
+            'ttype': 'integer',
+            'store': False,
+            'compute': compute_function.replace('    ', ''),
```

Comment:
```suggestion
            'compute': dedent(compute_function).strip(),
```

with dedent imported as
```python
from textwrap import dedent
```

Reply:
Nice idea but I would like to keep the exact same behavior as the code in studio that generate the computed field https://github.com/odoo/enterprise/blob/15.0/web_studio/controllers/main.py#L979